src/background/Wallet/Wallet.ts

```diff
@@ -810,8 +813,9 @@ export class Wallet {
     spender: string;
   }>) {
     this.verifyInternalOrigin(context);
-    const networks = await networksStore.load();
+    const networks = await networksStore.load([chain]);
```

Comment:
I don't understand why we pass `[chain]` in these method. Isn't the network store expected to have them?

Reply:
This is one of the key points of the new system. If you need to work with a chain, it is better to load networks with this chain as a param.
If this chain is there, getter will not load anything, but in other cases you just don't need to think if it is here or not. This is especially important because we have 2 separate networks stores for UI and background. That's why event is UI knows about the networks, this can be the first time background gets it.